### ISSUE TITLE
Cross Validation

### ISSUE THREAD

**Issue by Janecxl** [Janecxl/Notes#18]
## **Bias-Variance Tradeoff**
![](https://latex.codecogs.com/svg.latex?&space;E(\hat{\theta}-\theta)^2=(E\hat{\theta}-\theta)^2+Var(\hat{\theta})) 
MSE = bias^2 + variance

- Less flexible model: bias high, variance low
- More flexible model: bias low, variance high (overfitting issue)

## **What is Cross-Validation 🤔**
**Cross-Validation** is a resampling method that uses different portions of the data to test and train a model on different iterations.  It is a way to estimate the test error. 

## **Validation Set**
1. Randomly split dataset into two subsets
2. Use training set to fit the model
3. Use validation set  to estimate the test error

## **Leave-One-Out Cross-Validation (LOOCV)**

1. Split n data points into a training set of n−1 points and a validation set of 1 point
2. Consider all n possible ways of splitting 
3. Estimate test error by averaging
![](https://latex.codecogs.com/svg.latex?&space;CV=\frac{1}{n}\sum^n_{i=1}MSE_{i}) 

## **k-Fold Cross-Validation**
1. Randomly split the dataset into k subsets (folds) of equal size
2. Use (k − 1) subsets to fit model
3. Use the remaining 1 subset as the validation set
4. Estimate test error by averaging
![](https://latex.codecogs.com/svg.latex?&space;CV=\frac{1}{k}\sum^k_{i=1}MSE_{i}) 

Faster than LOOCV.

## **Reference 📖**
Cornell ORIE4740: Statistical Data Mining slides
